services/ResetDemo.php

```diff
@@ -0,0 +1,99 @@
+<?php
+// Service to initialize/reset demo database. Handles creating MySQL user "gradeplusclient", creating "gradeplus" database, creating and filling "login" table.
+if ($_POST["authorize"] == "gradeplus") {
+    try {
+        // Initialize/Reset Demo Database
+        // Connect to MySQL as admin
+        $conn = mysqli_connect('localhost', 'root', '');
+        if (!$conn){
+            error_log("Connection to MySQL as admin failed: " . mysqli_connect_error());
+        }
+
+        // Check if gradeplusclient user exists
+        $checkUserSql = "SELECT EXISTS(SELECT 1 FROM mysql.user WHERE user = 'gradeplusclient')";
+        $result = mysqli_query($conn, $checkUserSql);
+        $row = mysqli_fetch_array($result);
+        // Create user and give privileges if it does not exist
+        if ($row[0] == 0){
+            $createUserSql = "CREATE USER 'gradeplusclient'@'localhost' IDENTIFIED BY 'gradeplussql'";
+            $result = mysqli_query($conn, $createUserSql);
+            if (!$result){
+                error_log("Create user query failed: " . mysqli_error());
```

Comment:
I am getting 7 similar problems here in the mysql_error() function. Does this happen on your side?
I think it needs `$conn`  as an argument.

Reply:
Yep, that's my bad. mysqli_connect_error doesn't take any parameters, but mysqli_error needs the connection. I'll fix that.